0000-prelude-2024-future.md

```diff
@@ -0,0 +1,94 @@
+- Feature Name:  `prelude_2024_future`
+- Start Date:  2023-10-05
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC describes the inclusion of the `Future` and `IntoFuture` traits in the 2024 edition prelude.
+
+# Motivation
+[motivation]: #motivation
+
+When an `async fn` is desugared we obtain an anonymous type with a signature of `impl Future`. In order to use this type we must first be able to _name_ it, which can only be done if the  `Future` trait in scope. Currently this can be a little inconvenient since it requires `std::future::Future` to be manually imported.
+
+Most other reifications of control-flow effects have their respective types and traits included in the prelude header. To support iteration we include both the `Iterator` and `IntoIterator` traits in the prelude. To support fallibility we include both `Option` and `Result` in the prelude. This RFC proposes we include both `Future` and `IntoFuture` to match.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Let's say someone wrote an async function which takes a future and operates on it. For example, it could be something like this:
```

Comment:
```suggestion
In Rust 2024, you can name and make use of the `Future` and `IntoFuture` traits without explicitly importing them, since they appear in the Rust 2024 prelude. For instance, you can write a function that accepts or returns an `impl Future`, or one that accepts any `impl IntoFuture`.

Let's say someone wrote an async function which takes a future and operates on it. For example, it could be something like this:
```